logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/team/SyncSelfTeamUseCase.kt

```diff
@@ -0,0 +1,22 @@
+package com.wire.kalium.logic.feature.team
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.team.TeamRepository
+import com.wire.kalium.logic.data.user.UserRepository
+import com.wire.kalium.logic.functional.Either
+import kotlinx.coroutines.flow.first
+
+interface SyncSelfTeamUseCase {
```

Comment:
I guess we can make it internal.

```suggestion
internal interface SyncSelfTeamUseCase {
```

These sync UseCases are not really supposed to be exposed to whoever is consuming Kalium, but rather for Kalium workers to orchestrate.